### DISCUSSION TITLE
Gradle native build always fails with "Outputting both native and JAR packages is not currently supported"

### DISCUSSION THREAD

**Discussion by languitar** [quarkusio/quarkus#40679] [Q&A] [answered]
In a new Gradle Quarkus project using 3.10.1, I am unable to produce a native executable:

```
❯ ./gradlew build -Dquarkus.native.enabled=true -Dquarkus.jar.enabled=false
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true -Dswing.defaultlaf=com.sun.java.swing.plaf.gtk.GTKLookAndFeel
Starting a Gradle Daemon, 2 stopped Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':quarkusBuild'.
> Outputting both native and JAR packages is not currently supported

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
```

This also happens when using `--native` instead.

Interestingly, the native executable is built perfectly when running `./gradlew testNative`.

Any ideas what could be wrong here?

[a comment by an automated account is not shown]

**Comment by dmlloyd** [comment, marked as answer]
The short answer is, if you're building for native, set `quarkus.package.jar.enabled=false` and that should get you past the error. N.b. `quarkus.jar.enabled` is not the correct config key.

The long answer is that the Gradle tooling (specifically) doesn't support multiple outputs very well and in fact the integration isn't as good as I'd like. I'd like to discuss a plan to improve matters but it doesn't look to be very simple, so for now I think the workaround is what we have.

**Reply by dmlloyd** [reply]
Also, it appears that there is a bug here because the wrong config key is indeed set by the Gradle runner. Will put up a patch.

**Reply by languitar** [reply]
Thanks, then also the documentation needs to be updated in case the jar has to be explicitly disabled. https://quarkus.io/guides/config#package-and-run-the-application doesn't contain explicit disabling. Moreover, some parts of the docs use `--native` and others use `-Dquarkus.native.enabled=true`, which should probably be unified?

**Comment by geoand** [comment]
I just stumbled upon this myself. @gsmet @aloubyansky I think we need to improve our docs and our the generated README of the project
